tools/windows-node-installer/test/e2e/aws_test.go

```diff
@@ -162,12 +162,22 @@ func getInstanceStatus(instanceId string) (string, error) {
 	}
 	result, err := ec2Svc.DescribeInstanceStatus(input)
 	if err != nil {
-		return "", fmt.Errorf("failed to obtain the Instance status: %v", err)
+		return "", fmt.Errorf("failed to obtain the Instance status with error: %v", err)
 	}
+	if result.InstanceStatuses == nil {
+		return "", fmt.Errorf("no instance statuses were returned")
+	}
+
 	// currently we are creating only a single instance which is going to be the
 	// zeroth entry in it.
 	if result.InstanceStatuses[0] == nil {
-		return "", fmt.Errorf("failed to obtain Instance status")
+		return "", fmt.Errorf("windows instance status is empty")
+	}
+	if result.InstanceStatuses[0].InstanceStatus == nil {
+		return "", fmt.Errorf("windows instance summary status is empty")
+	}
+	if result.InstanceStatuses[0].InstanceStatus.Status == nil {
+		return "", fmt.Errorf("windows instance status field is empty")
```

Comment:
Please use `InstanceStatus` and `InstanceStatus.Status` instead of your own terminology.